decaporg/decap-cms
# Support prefix & suffix for 3 widgets: string, number & boolean

This changes are about:
- refactor syntax `BooleanControl` static properties
- add `prefix` & `suffix` options
- update widget control style to match with new options
- update Boolean widget document



**Summary**

The update allows developers to display messages before (`prefix`) and after (`suffix`) the toggle. This design intends to provide a better information and look to users.

In my use case, I have a field to decide if I want to show a post in homepage. If I set "Show in homepage" string in label, the layout of input would be unbalance as the top small part is bigger than the main body. The toggle itself seems lonely as well. So if I can set "Visibility" in `label` option and "Show in homepage" in `prefix` option, it would be much sense.

**Test plan**



1. Run `yarn start`
2. Go to this path `collections/kitchenSink/entries/a-big-entry-with-all-the-things`
3. Expect to see the exact Boolean input like this screenshot:   
   ![image](https://github.com/decaporg/decap-cms/assets/68631458/ddb01c07-1829-4d48-930e-8d4cb703994e)

**Checklist**




**A picture of a cute animal (not mandatory but encouraged)**

Be hold! The amazing meowman!

[![meow](https://github.com/decaporg/decap-cms/assets/68631458/5e610ad0-fd00-4a30-9378-3499d103da3d)](https://www.reddit.com/r/Eyebleach/comments/14hfbbe/ʍoǝɯ)
